packages/muelu/src/Graph/UncoupledAggregation/MueLu_AggregationPhase1Algorithm_kokkos_def.hpp

```diff
@@ -266,132 +286,104 @@ namespace MueLu {
   template <class LocalOrdinal, class GlobalOrdinal, class Node>
   void AggregationPhase1Algorithm_kokkos<LocalOrdinal, GlobalOrdinal, Node>::
   BuildAggregatesDistance2(const LWGraph_kokkos& graph, Aggregates_kokkos& aggregates,
-      std::vector<unsigned>& aggStat, LO& numNonAggregatedNodes, LO maxAggSize) const
+                           Kokkos::View<unsigned*, typename MueLu::LWGraph_kokkos<LO, GO, Node>::
+                           local_graph_type::device_type::memory_space>& aggStatView,
+                           LO& numNonAggregatedNodes, LO maxAggSize,
+                           Kokkos::View<LO*, typename MueLu::LWGraph_kokkos<LO, GO, Node>::
+                           local_graph_type::device_type::memory_space>& colorsDevice,
+                           LO& numColors) const
   {
+    typedef typename MueLu::LWGraph_kokkos<LO, GO, Node>::local_graph_type graph_t;
+    typedef typename graph_t::device_type::memory_space memory_space;
+    typedef typename graph_t::device_type::execution_space execution_space;
+
     const LO  numRows = graph.GetNodeNumVertices();
     const int myRank  = graph.GetComm()->getRank();
 
-    ArrayRCP<LO> vertex2AggId = aggregates.GetVertex2AggId()->getDataNonConst(0);
-    ArrayRCP<LO> procWinner   = aggregates.GetProcWinner()  ->getDataNonConst(0);
+    auto vertex2AggIdView = aggregates.GetVertex2AggId()->template getLocalView<memory_space>();
+    auto procWinnerView = aggregates.GetProcWinner()    ->template getLocalView<memory_space>();
 
+    LO numNonAggregatedNodesSerial = numNonAggregatedNodes;
```

Comment:
I'm getting an declared but unused warning for numNonAggregatedNodesSerial (CUDA build on white)

Reply:
@brian-kelley: OK, I took care of that, you should not have any problem compiling any more.
I have also found a good solution for the memory access problem. I am now working on phase 2b aggregation where I am allocating too much L0 memory apparently...